src/lib/tls/tls_session.cpp

```diff
@@ -428,15 +428,15 @@ Session Session::decrypt(std::span<const uint8_t> in, const SymmetricKey& key) {
          throw Decoding_Error("Encrypted session too short to be valid");
       }
 
-      // TODO: replace those raw pointers with sub-spans into `in`
-      const uint8_t* magic = in.data();
-      const uint8_t* key_name = magic + TLS_SESSION_CRYPT_MAGIC_LEN;
-      const uint8_t* key_seed = key_name + TLS_SESSION_CRYPT_KEY_NAME_LEN;
-      const uint8_t* aead_nonce = key_seed + TLS_SESSION_CRYPT_AEAD_KEY_SEED_LEN;
-      const uint8_t* ctext = aead_nonce + TLS_SESSION_CRYPT_AEAD_NONCE_LEN;
+      size_t inoff = 0;
+      const auto key_name = in.subspan(inoff += TLS_SESSION_CRYPT_MAGIC_LEN).data();
+      const auto key_seed = in.subspan(inoff += TLS_SESSION_CRYPT_KEY_NAME_LEN).data();
+      const auto aead_nonce = in.subspan(inoff += TLS_SESSION_CRYPT_AEAD_KEY_SEED_LEN).data();
+      const auto ctext = in.subspan(inoff += TLS_SESSION_CRYPT_AEAD_NONCE_LEN).data();
+      BOTAN_ASSERT(inoff < in.size(), "input consistency check");
```

Comment:
Actually, I think we want to use the `BufferSlicer` helper here as it keeps track of what is here called `inoff` internally.

Like so:

```C++
BufferSlicer s(in);
const auto magic =  s.take(TLS_SESSION_CRYPT_MAGIC_LEN);
const auto key_name = s.take(TLS_SESSION_CRYPT_KEY_NAME_LEN);
...
```